### PR TITLE
feat(Reserve):  Reserve can burn RUN to reduce shortfall

### LINKED ISSUE #5299: Reserve contract covers vault shortfalls
## What is the Problem Being Solved?
The Inter Protocol relies on successful liquidation to ensure the stable token remains over collateralized.  In cases where individual liquidations do not cover the outstanding debt for any reason, the protocol must track the outstanding balance so that it can be made whole through later operations.  This ticket allows the Reserve contract to both track the ongoing shortfall and to burn IST to reduce it.

## Description of the Design
Shortfalls that occur during liquidation are passed to the reserve contract which manages a shortfall balance.  This logic must survive as liquidation contract is changed.

The reserve contract must burn IST to reduce shortfall.  Suggest adding method that can be called by governance which takes IST from the Reserve's IST purse to handle this.

Reserve's IST purse is funded through execution fees and excess reward flows as described in #4399, but it must also be able to accept additional ad-hoc deposits as needed.  I.e., governance could decide to sell other assets to and fund the Reserve IST purse for the purpose of burning to reduce shortfall.

## Security Considerations
The liquidation shortfall which occurs during execution of the liquidation contract within the vault manager must be passed correctly to the reserve to ensure shortfalls cannot fail to be tracked

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
closes: #5299

## Description

Add an ability (controlled by governance) to burn RUN from the Reserve to reduce shortfalls from liquidations.

### Security Considerations

Governance & economic

### Documentation Considerations

We haven't started documenting the capabilities accessible to the econ committee.  This is one.

### Testing Considerations

Add tests so show that the shortfall is reduced.

### CODE REVIEW

File: packages/run-protocol/src/reserve/assetReserve.js [hunk 1]
```diff
@@ -222,6 +222,19 @@ const start = async (zcf, privateArgs) => {
     zcf.reallocate(offerToSeat, collateralSeat);
   };
 
+  const burnRUNToReduceShortfall = reduction => {
```
Comment (turadg) [inline packages/run-protocol/src/reserve/assetReserve.js:225]: RUN is all-caps even in function names?

File: packages/run-protocol/src/reserve/assetReserve.js [hunk 2]
```diff
@@ -222,6 +222,19 @@ const start = async (zcf, privateArgs) => {
...
+    reduction = AmountMath.coerce(runBrand, reduction);
+    const runKeyword = keywordForBrand.get(runBrand);
+    const runBalance = collateralSeat.getAmountAllocated(runKeyword);
+    const amountToBurn = AmountMath.min(reduction, runBalance);
+    if (AmountMath.isEmpty(amountToBurn)) {
+      return;
+    }
+
+    runMint.burnLosses(harden({ [runKeyword]: amountToBurn }), collateralSeat);
```
Comment (turadg) [inline packages/run-protocol/src/reserve/assetReserve.js:234]: does this still allow that the assertReserve contract's keywords are private and meaningless?

Comment (Chris-Hibbert) [reply packages/run-protocol/src/reserve/assetReserve.js:234]: it should.  Just above this, `runKeyword` is set using `const runKeyword = keywordForBrand.get(runBrand);`, so whatever we use for the keywords should still be in `keywordForbrand`.

File: packages/run-protocol/src/reserve/assetReserve.js (hunk 1 above)
Comment (Chris-Hibbert) [reply packages/run-protocol/src/reserve/assetReserve.js:225]: That was the policy I understood. I wouldn't mind if that's not the rule.

Comment (turadg) [reply packages/run-protocol/src/reserve/assetReserve.js:225]: can wait for https://github.com/Agoric/agoric-sdk/issues/4800

File: packages/run-protocol/src/reserve/assetReserve.js (hunk 2 above)
Comment (turadg) [reply packages/run-protocol/src/reserve/assetReserve.js:234]: cool. I wasn't sure how keywords interacted with burnLosses but I see they're ignored https://github.com/Agoric/agoric-sdk/blob/fd44fca1ae25f9b17c34404bc03a7e79bdb4982b/packages/zoe/src/contractFacet/zcfZygote.js#L174-L175

**Review by turadg — APPROVED** [review]

**Pull request merged**